kotlin/vector-tiled-layer-custom-style/src/main/java/com/esri/arcgisruntime/sample/vectortiledlayercustomstyle/MainActivity.kt

```diff
@@ -0,0 +1,293 @@
+        val task = ExportVectorTilesTask(portalItem)
+
+        val exportVectorTilesJob = task.exportStyleResourceCache(getExternalFilesDir(null)?.path + "/"+portalItem.itemId)
+        exportVectorTilesJob.addJobDoneListener {
+            if (exportVectorTilesJob.status == Job.Status.SUCCEEDED) {
+                setResourceAndVectorTileCache(exportVectorTilesJob.result.itemResourceCache)
+            } else {
+                Toast.makeText(this, "Error reading cache: " + exportVectorTilesJob.error.message, Toast.LENGTH_LONG).show()
+            }
+        }
+        exportVectorTilesJob.start()
+
+    }
+
+    private fun setResourceAndVectorTileCache(itemResourceCache: ItemResourceCache){
+        //Loads the vector tile layer cache.
+        val vectorTileCache = VectorTileCache(getExternalFilesDir(null)?.path + "/dodge_city.vtpk")
+        vectorTileCache.loadAsync()
+        vectorTileCache.addDoneLoadingListener{
+            if(vectorTileCache.loadError != null)
+                Log.e("VectorTileCache: " , vectorTileCache.loadError.message.toString())
+            else {
+                // Loads the layer based on the vector tile cache and the style resource.
+                val layer = ArcGISVectorTiledLayer(vectorTileCache, itemResourceCache)
+                layer.addDoneLoadingListener {
+                    if (layer.loadError != null)
+                        Log.e("VectorTiledLayer: ", layer.loadError.toString())
+                    else
+                        Log.d("VectorTiledLayer:" , "Loaded successfully")
+                }
+
+                // OfflineItemIDs uses WGS-84 as a spatial ref
+                val viewpoint = Viewpoint(Point(-100.01766, 37.76528, SpatialReferences.getWgs84()), 100000.0)
+                setMap(layer, viewpoint)
+            }
+        }
+    }
+
+    /**
+     * Set the map using the layer and the viewpoint.
+     */
+    private fun setMap(layer: ArcGISVectorTiledLayer, viewpoint: Viewpoint){
+        //Reset the map to release resources
+        mapView.map = null
+        //mapView.map.basemap = null
+
+        // Assign a new map created from the base layer.
+        val basemap = Basemap(layer.copy())
+        val map = ArcGISMap(basemap)
+        mapView.map = map
+        //Set viewpoint without animation.
+        mapView.setViewpoint(viewpoint)
+    }
+
+    /**
+     * Sets the adapter and listens for a item selection to update the
+     * MapView with the selected layer.
+     */
+    private fun setUpSpinner() {
+        val customDropDownAdapter = CustomDropDownAdapter(this)
+        spinner.adapter = customDropDownAdapter
+        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener{
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+
+                // Sets the selected itemID to either the Online/Custom ID.
+                currentItemID = when (position) {
+                    4 -> {
```

Comment:
I understand what you're doing here, but a comment to explain why you're only checking for positions 4 and 5 would be helpful here